app/src/main/java/com/willowtree/vocable/CategoriesUseCase.kt

```diff
@@ -29,15 +29,40 @@ class CategoriesUseCase(
 
     override suspend fun getCategoryById(categoryId: String): Category {
         return storedCategoriesRepository.getCategoryById(categoryId)?.asCategory()
-            ?: presetCategoriesRepository.getCategoryById(categoryId)
+            ?: presetCategoriesRepository.getCategoryById(categoryId)?.let { if (it.hidden) null else it }
             ?: throw IllegalArgumentException("Category with id $categoryId not found")
     }
 
     override suspend fun updateCategoryName(
         categoryId: String,
         localizedName: LocalesWithText
     ) {
-        legacyCategoriesAndPhrasesRepository.updateCategoryName(categoryId, localizedName)
+        val storedCategory = storedCategoriesRepository.getCategoryById(categoryId)
+        if (storedCategory != null) {
+            storedCategoriesRepository.upsertCategory(
+                Category.StoredCategory(
+                    storedCategory.categoryId,
+                    localizedName,
+                    storedCategory.hidden,
+                    storedCategory.sortOrder
+                )
+            )
+        } else {
+            val presetCategory = presetCategoriesRepository.getCategoryById(categoryId)
+            if (presetCategory != null) {
+                presetCategoriesRepository.hidePresetCategory(categoryId)
+                storedCategoriesRepository.upsertCategory(
+                    Category.StoredCategory(
+                        presetCategory.categoryId,
+                        localizedName,
+                        presetCategory.hidden,
+                        presetCategory.sortOrder
+                    )
+                )
+            } else {
+                throw IllegalArgumentException("Category with id $categoryId not found")
+            }
+        }
```

Comment:
Just curious- is this a solved problem? Throw exception vs. return result (vs silently swallow/log errors 👀)

If you've got any sage words or content you could point me to about this I'd appreciate it. Otherwise I can just research it a little and discuss with the smart folks at office hours tomorrow

Reply:
My stance on exceptions [is generally aligned with this guy](https://james1345.medium.com/exceptions-result-types-and-kotlin-235af74e43e0).

In this case, this method being called with a category ID that cannot be found is *not* an expected state of our application. It could be argued that we don't want the app to crash in this case, but I'd also argue that there's a benefit to a crash in that it's very obvious something has gone wrong, and requires the least lift _right now_.

I think at the least we'd want a result return type, so that we could communicate to the caller that things broke, so that it can react accordingly. If we fail silently, we may leave the user stuck in an invalid state, which is worse than a crash IMO. 

Granted, as I type this, perhaps a crash in Vocable _is_ worst case, because the user may not be able to launch Vocable again themselves if they have limited mobility.

What are your thoughts?